### PR TITLE
Dockerize connect jenkins testing

### PR DESCRIPTION
**Pull request by wstephenson**

### CODE REVIEW

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
http://river.suse.de/job/SCC-Connect-Prophet/150523/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-build-image.
( Failure: Merged 0cd80183481bde3d2b9f12531d5db6b9c279a1e0 into 5c69e45b0a9f39afff387b2c607e22900f85e60e )

**Comment by digitaltom** [comment]
The image should mount `/var/tmp/osbuild-packagecache/` to avoid repeated package downloads.

File: Dockerfile [hunk 1]
```diff
@@ -17,4 +17,6 @@ WORKDIR /tmp/connect
 RUN bundle -j8
 ADD . /tmp/connect
 RUN chown -R nobody /tmp/connect
-RUN chmod +x /tmp/connect/docker/runall.sh
+RUN chmod +x /tmp/connect/docker/prophet_tests.sh /tmp/connect/docker/integration.sh
+ADD .oscrc /root/.oscrc
+ADD .regcode /root/.regcode
```
Comment (wstephenson) [inline Dockerfile]: Do you mean why don't we fetch them from gaffer at runtime inside the image? or why don't we RUN wget to get them? The second option is cached too aggressively by docker build.

**Comment by wstephenson** [comment]
.oscrc now sets the packagecachedir to a different shared volume mounted at /oscbuild/packagecache

File: docker/integration.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/bin/sh -xe
+if [ -z $PRODUCT ]
+then
+  echo "PRODUCT env var for integration testing not set!"
+  exit 1
+fi
+
+rake build[$PRODUCT] &&
+zypper --non-interactive --no-gpg-checks in /oscbuild/$PRODUCT-x86_64/home/abuild/rpmbuild/RPMS/x86_64/* &&
+cucumber /tmp/connect/features/activation.feature /tmp/connect/features/activation_errors.feature
```
Comment (wstephenson) [inline docker/integration.sh]: I kept the test selection from the previous implementation.  We can run all the tests on every SP, but they will require a lot more work to make them version independent, as I have done with the activation tests.

**Comment by digitaltom** [comment]
Yep, caching works now :car:

**Comment by digitaltom** [comment]
It should, I can see `packagecachedir = /oscbuild/packagecache` inside of `~/.oscrc`. 
But still the packages are downloaded to `/var/tmp/osbuild-packagecache/`.

File: Dockerfile (hunk 1 above)
Comment (digitaltom) [inline Dockerfile]: I would go for getting it added in `RUN`, because we can save time on every run, and those files don't change often.

**Comment by digitaltom** [comment]
Probably `packagecachedir` needs to be moved up to the `General` section

File: docker/integration.sh (hunk 2 above)
Comment (digitaltom) [inline docker/integration.sh]: at least help, localization and version.feature should work, to make it working completely we should add a card to the backlog.

**Comment by wstephenson** [comment]
.oscrc updated, next build should take it and cache correctly.

File: Dockerfile (hunk 1 above)
Comment (wstephenson) [inline Dockerfile]: NB fetching them from gaffer into the image allows the image to be used standalone without knowing Jenkins config <- Tom

**Comment by wstephenson** [comment]
Caching confirmed working

File: Dockerfile (hunk 1 above)
Comment (wstephenson) [inline Dockerfile]: Reverted to wget in RUN, for the above reasons.

File: Dockerfile [hunk 3]
```diff
@@ -1,11 +1,11 @@
 FROM registry.scc.suse.de/sles12_base
 
+# Remember to drop docker caches if any of these change
 RUN zypper --non-interactive ar http://download.suse.de/ibs/SUSE:/SLE-12:/GA/standard/ SLE-12-standard &&\
     zypper --non-interactive ar -f http://download.suse.de/ibs/SUSE:/SLE-12:/Update/standard/ SLE-12-update-standard &&\
+    zypper --non-interactive ar -f http://download.opensuse.org/repositories/openSUSE:/Tools/SLE_12/ opensuse-tools && \
```
Comment (wstephenson) [inline Dockerfile]: Done

File: Dockerfile [hunk 4]
```diff
@@ -17,4 +17,6 @@ WORKDIR /tmp/connect
 RUN bundle -j8
 ADD . /tmp/connect
 RUN chown -R nobody /tmp/connect
-RUN chmod +x /tmp/connect/docker/runall.sh
+RUN chmod +x /tmp/connect/docker/prophet_tests.sh /tmp/connect/docker/integration.sh
```
Comment (wstephenson) [inline Dockerfile]: Done.

File: Dockerfile [hunk 5]
```diff
@@ -17,4 +17,6 @@ WORKDIR /tmp/connect
 RUN bundle -j8
```
Comment (wstephenson) [inline Dockerfile]: Done.

File: Dockerfile.sp2 [hunk 6]
```diff
@@ -18,4 +18,6 @@ WORKDIR /tmp/connect
 RUN bundle -j8
 ADD . /tmp/connect
 RUN chown -R nobody /tmp/connect
-RUN chmod +x /tmp/connect/docker/runall.sh
+RUN chmod +x /tmp/connect/docker/prophet_tests.sh /tmp/connect/docker/integration.sh
+ADD .oscrc /root/.oscrc
+ADD .regcode /root/.regcode
```
Comment (wstephenson) [inline Dockerfile.sp2]: Done.

File: docker/integration.sh (hunk 2 above)
Comment (wstephenson) [inline docker/integration.sh]: Card added and these tests are added to the integration.sh set.

**Comment by digitaltom** [comment]
@wstephenson i think the whole kitchen/ folder can be removed.

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
http://river.suse.de/job/SCC-Connect-Prophet/151500/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests.
( Failure: Merged 1ed13db4d268eb759389719d721ae94a250b9181 into 5c69e45b0a9f39afff387b2c607e22900f85e60e )

File: Dockerfile.sp1 [hunk 7]
```diff
@@ -2,12 +2,12 @@ FROM registry.scc.suse.de/suse/sles12:sp1
 
 # Remember to drop docker caches if any of these change
 RUN zypper --non-interactive --no-refresh rm container-suseconnect &&\
-    zypper --non-interactive ar http://download.suse.de/ibs/SUSE/Products/SLE-SERVER/12-SP1/x86_64/product/ SLE-12-SP1-standard &&\
-    zypper --non-interactive ar -f http://download.suse.de/ibs/SUSE/Updates/SLE-SERVER/12-SP1/x86_64/update/ SLE-12-SP1-updates &&\
-    zypper --no-gpg-checks --non-interactive ar http://download.suse.de/ibs/SUSE/Products/SLE-SDK/12-SP1/x86_64/product/ sle12sp1sdk &&\
-    zypper --non-interactive ar -f http://download.opensuse.org/repositories/openSUSE:/Tools/SLE_12_SP1/ opensuse-tools && \
+    zypper ar -f http://download.suse.de/ibs/SUSE/Products/SLE-SERVER/12-SP1/x86_64/product/ SLE-12-SP1-standard &&\
```
Comment (digitaltom) [inline Dockerfile.sp1]: The pool repository is static, it does not need the refresh flag.

File: Dockerfile.sp1 [hunk 8]
```diff
@@ -2,12 +2,12 @@ FROM registry.scc.suse.de/suse/sles12:sp1
...
+    zypper ar -f http://download.suse.de/ibs/SUSE/Updates/SLE-SERVER/12-SP1/x86_64/update/ SLE-12-SP1-updates &&\
+    zypper --no-gpg-checks ar -f http://download.suse.de/ibs/SUSE/Products/SLE-SDK/12-SP1/x86_64/product/ sle12sp1sdk &&\
```
Comment (digitaltom) [inline Dockerfile.sp1]: This repo is static, it does not need a refresh flag

File: Dockerfile.sp2 [hunk 9]
```diff
@@ -2,11 +2,11 @@ FROM registry.scc.suse.de/suse/sles12:sp2
 
 # Remember to drop docker caches if any of these change
 RUN zypper --non-interactive ar http://download.suse.de/ibs/SUSE/Products/SLE-SERVER/12-SP2/x86_64/product/ SLE-12-SP2-standard &&\
-    zypper --non-interactive ar -f http://download.suse.de/ibs/SUSE/Updates/SLE-SERVER/12-SP2/x86_64/update/ SLE-12-SP2-updates &&\
-    zypper --no-gpg-checks --non-interactive ar http://download.suse.de/ibs/SUSE/Products/SLE-SDK/12-SP2/x86_64/product/ sle12sp2sdk &&\
-    zypper --non-interactive ar -f http://download.opensuse.org/repositories/openSUSE:/Tools/SLE_12_SP1/ opensuse-tools && \
+    zypper ar -f http://download.suse.de/ibs/SUSE/Updates/SLE-SERVER/12-SP2/x86_64/update/ SLE-12-SP2-updates &&\
+    zypper ar -f http://download.suse.de/ibs/SUSE/Products/SLE-SDK/12-SP2/x86_64/product/ sle12sp2sdk &&\
```
Comment (digitaltom) [inline Dockerfile.sp2]: This repo is static, it does not need a refresh flag

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
http://river.suse.de/job/SCC-Connect-Prophet/151903/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests.
( Failure: Merged 1ed13db4d268eb759389719d721ae94a250b9181 into dcc825a15596a4da65e7e2debca72940de2273b8 )

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
http://river.suse.de/job/SCC-Connect-Prophet/152070/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests.
( Failure: Merged 6cc21d32cfae7c425091eb4d56f4ae2bc291f58b into dcc825a15596a4da65e7e2debca72940de2273b8 )

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
http://river.suse.de/job/SCC-Connect-Prophet/152086/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests.
( Failure: Merged 982b54bb5fc191ca8d16fa58398072dd72952db8 into dcc825a15596a4da65e7e2debca72940de2273b8 )

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
http://river.suse.de/job/SCC-Connect-Prophet/152087/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests.
( Failure: Merged a4e249b7f6fcdf222a0d05589139b32b4b9e0a43 into dcc825a15596a4da65e7e2debca72940de2273b8 )

File: Dockerfile.sp1 (hunk 7 above)
Comment (wstephenson) [inline Dockerfile.sp1]: Fixed

File: Dockerfile.sp2 (hunk 9 above)
Comment (wstephenson) [inline Dockerfile.sp2]: Fixed

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
http://river.suse.de/job/SCC-Connect-Prophet/152090/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests.
( Failure: Merged 8c51d25d3e6b01d64e9e63b3df1caf9cfda780bb into dcc825a15596a4da65e7e2debca72940de2273b8 )

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
http://river.suse.de/job/SCC-Connect-Prophet/152091/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests.
( Failure: Merged c5f7ab45a569f8b561a6ff088d2ca105c477d9b7 into dcc825a15596a4da65e7e2debca72940de2273b8 )

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
http://river.suse.de/job/SCC-Connect-Prophet/152387/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests.
( Failure: Merged c5f7ab45a569f8b561a6ff088d2ca105c477d9b7 into 95d81d699cf674ca28fd467b0fa32c6eeeb81481 )

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
http://river.suse.de/job/SCC-Connect-Prophet/152388/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests.
( Failure: Merged 1a07efc9580c21571c636f2dab8cc730fa10e254 into 95d81d699cf674ca28fd467b0fa32c6eeeb81481 )

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
http://river.suse.de/job/SCC-Connect-Prophet/152410/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
( Success: Merged 70e6324057d7c8282874fc8ca666457b2f441a0c into 95d81d699cf674ca28fd467b0fa32c6eeeb81481 )

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
http://river.suse.de/job/SCC-Connect-Prophet/152490/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
( Success: Merged fa8bdcdb137085d424427e5b037864c8eb6b3034 into 95d81d699cf674ca28fd467b0fa32c6eeeb81481 )

**Pull request merged**